### DISCUSSION TITLE
Differentiating Production vs Development Environment Databases

### DISCUSSION THREAD

**Discussion by douglasrcjames** [prisma/prisma#20661] [Q&A] [answered]
### Question

I am using Prisma with Next.js 13 and Vercel's Postgres storage solution. I am stuck on how I can have one Next.js code base that will automatically reference the development database when on localhost or any test environment and then also reference the production database only on the production side. How is this achieved with this stack? 

With Firebase I would just reference two different Firebase projects, differentiating which API config keys to use via something like `process.env.NODE_ENV === "development" ? DEV_KEYS : PROD_KEYS`. However, I cannot see how I can have a conditional statement in the `schema.prisma` file? Is there somewhere else in a Next.js build where I can reference two different postgres databases between live and test environments? Anything helps here, thanks!

```prisma
generator client {
  provider = "prisma-client-js"
  previewFeatures = ["fullTextSearch"]
}

datasource db {
  provider = "postgresql"
  url = env("POSTGRES_PRISMA_URL") // uses connection pooling
  directUrl = env("POSTGRES_URL_NON_POOLING") // uses a direct connection
}

model User {
    id          String  @id @default(cuid())
    name        String?
    email       String  @unique
    phone       String?
    createdAt   DateTime  @default(now()) @map(name: "created_at")
    updatedAt   DateTime  @updatedAt @map(name: "updated_at")
}

```

**Comment by ludralph** [comment, marked as answer]
Hi @douglasrcjames 👋 

Prisma schema files don't support conditional logic directly. You can achieve conditional database configuration based on the environment by utilizing environment variables. One solution is to have multiple .env files, which each represent different environments. See this [section](https://www.prisma.io/docs/guides/development-environment/environment-variables/using-multiple-env-files) of the documentation. 

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by douglasrcjames** [reply]
This pretty much does yes, I was informed on this thread https://github.com/prisma/prisma/issues/2443#issuecomment-1677359048 to pretty much use `dotenv` with a command like `dotenv -e .env.development.local -- npx prisma generate`
